app/src/main.js

```diff
@@ -1,46 +1,56 @@
 const serverless = require('serverless-http');
 const express = require('express');
-const morgan = require('morgan');
-const morganJson = require('morgan-json');
 const superagent = require('superagent');
 
-const config = require('./config/loader').load();
-const smmtClient = require('./smmt/client').create(superagent, config);
+const smmtConfig = require('./config/smmt').load();
+const smmtClient = require('./smmt/client').create(superagent, smmtConfig);
+const logger = require('./logger/createLogger').create();
 
 const app = express();
 app.disable('x-powered-by');
 
-const logFormat = morganJson(':method :url :status :res[content-length] bytes :response-time ms');
-app.use(morgan(logFormat));
-
 app.get('/recalls', (req, res) => {
+  logger.debug(req, 'Received request.');
+
   const { make, vin } = req.query;
 
   if (make && vin) {
-    const result = smmtClient.vincheck(make, vin);
-
-    result.then((recall) => {
-      if (recall.success) {
-        res.status(200)
+    smmtClient.vincheck(make, vin)
+      .then((recall) => {
+        if (recall.success) {
+          logger.info({ context: { make, vin, recall } }, 'Recall fetched successfully.');
+          res.status(200)
+            .send({
+              status_description: recall.description,
+              vin_recall_status: recall.status,
+              last_update: recall.lastUpdate,
+            });
+        } else {
+          const context = {
+            make, vin, errors: recall.errors,
+          };
+          logger.error({ context }, 'Recall fetching from SMMT failed.');
```

Comment:
Please include response code from SMMT whenever error occurs. This will give a lot of information should unexpected event occurs.

Reply:
They are included:
```
{
    "name": "NotSet",
    "env": "NotSet",
    "functionName": "vehicle-recalls-api-dev-app",
    "functionVersion": "$LATEST",
    "hostname": "ip-10-12-118-227",
    "pid": 1,
    "level": 20,
    "context": {
        "recall": {
            "status": 402,
            "status_description": "Bad Request - Invalid Marque",
            "vin": "AISfgdfgdhdhdf",
            "vin_recall_status": "",
            "last_update": ""
        }
    },
    "msg": "Invalid SMMT marque.",
    "time": "2017-11-06T11:36:05.203Z",
    "v": 0
}
```